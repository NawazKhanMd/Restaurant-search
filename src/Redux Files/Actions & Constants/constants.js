export const home_action_types = {
    get_cities_success: 'get_cities_success',
    get_cities_fail: 'get_cities_success',
    get_restaurants_success: 'get_restaurants_success',
    get_restaurants_fail: 'get_restaurants_fail',
    filter_restaurants:'filter_restaurants',
    loading:'loading'
} 