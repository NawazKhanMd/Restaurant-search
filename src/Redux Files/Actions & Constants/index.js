export * from './home_actions';