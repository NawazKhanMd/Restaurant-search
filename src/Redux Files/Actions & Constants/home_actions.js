import { home_action_types as action_types } from '../Actions & Constants/constants'
import {restApiUrls,RequestAPI} from '../../Utils/fetchInterceptor'
export const getCities = () => (dispatch) => {
  dispatch(toogleLoader(true))
  let fetchObj = {
    url: restApiUrls.CitiesURL,
    method: 'GET'
}
return RequestAPI(fetchObj).then(resp => {
  dispatch(toogleLoader(false))
    if (resp.type == 'success') {
        dispatch({
          type:action_types.get_cities_success,
          payload:resp.data
        })
    } else {
      dispatch(toogleLoader(false))
    }
})
}
export const toogleLoader = (flag) => (dispatch)=>{
  dispatch({
    type:action_types.loading,
    flag:flag
  })
}
export const filterRestaurants= (address)=>(dispatch)=>{
  dispatch({
    type:action_types.filter_restaurants,
    address:address
  })
}

export const getRestaurants = (City) => (dispatch) => {
  dispatch(toogleLoader(true))
  let fetchObj = {
    url: restApiUrls.RestaurantURL.replace('#{CITY}#',City),
    method: 'GET'
}
return RequestAPI(fetchObj).then(resp => {
  dispatch(toogleLoader(false))
    if (resp.type == 'success') {
        dispatch({
          type:action_types.get_restaurants_success,
          payload:resp.data
        })
    } else {
      dispatch(toogleLoader(false))
    }
})
}

