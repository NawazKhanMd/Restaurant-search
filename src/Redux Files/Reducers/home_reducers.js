import {home_action_types as atn_typ} from '../Actions & Constants/constants'

export const initialState = {
    Cities: [],
    Restaurants:[],
    SavedRestaurants:[],
    Address:[],
    loading:false,
};


export const home = (state = initialState, action) => {

    switch (action.type) {
        case atn_typ.get_cities_success:
            state.Cities  = action.payload.data.cities.reduce((acc,val)=>{
                let obj ={label:val,value:val};
                acc.push(obj);
                return acc;
            },[])
            return {...state}
        case atn_typ.get_restaurants_success:
            state.Restaurants = action.payload.data.restaurants
            state.SavedRestaurants = action.payload.data.restaurants
            state.Address = action.payload.data.restaurants.reduce((acc,val)=>{
                let obj ={label:val.address,value:val.address};
                acc.push(obj);
                return acc;
            },[])
        return {...state}
        case atn_typ.filter_restaurants:
            if(action.address == ''){
                state.Restaurants = [...state.SavedRestaurants]
            }else if(action.address == null){
                state.Restaurants = []
                state.SavedRestaurants = []
            }else{
                state.Restaurants = state.SavedRestaurants.filter(ele=>ele.address == action.address)
            }
            
            return {...state}
       case atn_typ.loading:
           state.loading = action.flag
           return {...state}
        default:
            return state
    }
}
