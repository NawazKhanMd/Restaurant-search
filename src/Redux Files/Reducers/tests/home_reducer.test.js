import { home, initialState } from '../home_reducers';
import { home_action_types } from '../../Actions & Constants/constants';
//import { dummyVariousContent } from '../types/contentTypes';

describe('Home Reducer', () => {
    it('Handles default case', () => {
        const state = home(undefined, {});
        expect(state).toEqual(initialState);
    });
    it('Handles load Restaurants ', () => {
        const data = {restaurants:[]}
        data.restaurants = [{
            address: "1 Benvenuto Place",
            area: "Toronto / SW Ontario",
            city: "Toronto",
            country: "CA",
            id: 21307,
            image_url: "https://www.opentable.com/img/restimages/21307.jpg",
            lat: 43.68207,
            lng: -79.40041,
            mobile_reserve_url: "http://mobile.opentable.com/opentable/?restId=21307",
            name: "Scaramouche Restaurant",
            phone: "4169618011",
            postal_code: "M4V 2L1",
            price: 4,
            reserve_url: "http://www.opentable.com/single.aspx?rid=21307",
            state: "ON"
        }]
        const action = { type: home_action_types.get_restaurants_success, payload: {data:data} };
        const state = home(undefined, action);
        expect(state).toMatchSnapshot();
    });
    // it('Handles load delays ', () => {
    //     const delays = [{ "cor": "9", "date": "04-30-18" }]
    //     let sorted_delays = delays.reduce((acc, val) => {
    //         acc[val.date] = val.cor;
    //         return acc
    //       }, {})
    //     const action = { type: home_action_types.get_delays_success, delays: sorted_delays };
    //     const state = home(undefined, action);
    //     expect(state).toMatchSnapshot();
    // });
    // it('Handles load streets ', () => {
    //     let street1 = {
    //         streetName: '148th Avenue Southeast',
    //         cor: 25,
    //         path: [
    //             {
    //                 'lat': 47.580725569999998,
    //                 'lng': -122.1412285
    //             },
    //             {
    //                 'lat': 47.585360440000002,
    //                 'lng': -122.1428461
    //             },
    //             {
    //                 'lat': 47.589032889999999,
    //                 'lng': -122.14273470000001
    //             },
    //             {
    //                 'lat': 47.596341019999997,
    //                 'lng': -122.142674
    //             },
    //             {
    //                 'lat': 47.60259559,
    //                 'lng': -122.14280979999999
    //             },
    //             {
    //                 'lat': 47.609862249999999,
    //                 'lng': -122.1429724
    //             },
    //         ]
    //     };
    //     const streets = [street1]
    //     const action = { type: home_action_types.get_streets_success, streets: streets };
    //     const state = home(undefined, action);
    //     expect(state).toMatchSnapshot();
    // });
});