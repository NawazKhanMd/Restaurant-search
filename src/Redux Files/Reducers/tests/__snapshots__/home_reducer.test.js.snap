// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Reducer Handles load Emissions  1`] = `
Object {
  "Address": Array [],
  "Cities": Array [],
  "Restaurants": Array [],
  "SavedRestaurants": Array [],
  "loading": false,
}
`;

exports[`Home Reducer Handles load Events  1`] = `
Object {
  "Address": Array [],
  "Cities": Array [],
  "Restaurants": Array [],
  "SavedRestaurants": Array [],
  "loading": false,
}
`;

exports[`Home Reducer Handles load Incidents  1`] = `
Object {
  "Address": Array [],
  "Cities": Array [],
  "Restaurants": Array [],
  "SavedRestaurants": Array [],
  "loading": false,
}
`;

exports[`Home Reducer Handles load Restaurants  1`] = `
Object {
  "Address": Array [
    Object {
      "label": "1 Benvenuto Place",
      "value": "1 Benvenuto Place",
    },
  ],
  "Cities": Array [],
  "Restaurants": Array [
    Object {
      "address": "1 Benvenuto Place",
      "area": "Toronto / SW Ontario",
      "city": "Toronto",
      "country": "CA",
      "id": 21307,
      "image_url": "https://www.opentable.com/img/restimages/21307.jpg",
      "lat": 43.68207,
      "lng": -79.40041,
      "mobile_reserve_url": "http://mobile.opentable.com/opentable/?restId=21307",
      "name": "Scaramouche Restaurant",
      "phone": "4169618011",
      "postal_code": "M4V 2L1",
      "price": 4,
      "reserve_url": "http://www.opentable.com/single.aspx?rid=21307",
      "state": "ON",
    },
  ],
  "SavedRestaurants": Array [
    Object {
      "address": "1 Benvenuto Place",
      "area": "Toronto / SW Ontario",
      "city": "Toronto",
      "country": "CA",
      "id": 21307,
      "image_url": "https://www.opentable.com/img/restimages/21307.jpg",
      "lat": 43.68207,
      "lng": -79.40041,
      "mobile_reserve_url": "http://mobile.opentable.com/opentable/?restId=21307",
      "name": "Scaramouche Restaurant",
      "phone": "4169618011",
      "postal_code": "M4V 2L1",
      "price": 4,
      "reserve_url": "http://www.opentable.com/single.aspx?rid=21307",
      "state": "ON",
    },
  ],
  "loading": false,
}
`;
